web/modules/sheaf/src/state.ts

```diff
@@ -27,30 +53,121 @@ export class SheafState {
     this.activeLines = getActiveLines(opts.view.state)
   }
 
-  private declare _value?: string
-
+  /** Retrieves the value of the editor. */
+  @Memoize()
   async value() {
-    if (this._value) return this._value
-    return (this._value = await textValue(this.doc))
+    return await textValue(this.doc)
   }
 
+  /** Retrieves the value of the editor as an `Uint8Array`. */
+  @Memoize()
   async buffer() {
     return await textBuffer(this.doc)
   }
 
+  /** Renders the editor with FTML. */
+  @Memoize()
+  async render() {
+    return await FTML.renderHTML(await this.value(), undefined, "draft")
```

Comment:
Are these methods memoized because the value doesn't change, or is Typescript invalidating the cached result if any of the fields update?